core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -475,6 +533,194 @@ object ConsumerGroupCommand extends Logging {
+              case ts => s"${ts}Z"
+            }
+            val zonedDateTime = ZonedDateTime.parse(datetime)
+            println(zonedDateTime)
+            val timestamp = zonedDateTime.toInstant.toEpochMilli
+            val topicPartition = new TopicPartition(assignment.topic.get, assignment.partition.get)
+            val logTimestampOffset = getLogTimestampOffset(topicPartition, timestamp)
+            logTimestampOffset match {
+              case LogTimestampOffsetResult.LogTimestampOffset(offset) => assignment -> offset
+              case LogTimestampOffsetResult.Unknown => null
+              case LogTimestampOffsetResult.Ignore => null
+            }
+          }
+        ).toMap
+      } else if(opts.options.has(opts.resetByDurationOpt)){
+        assignmentsToReset.map(
+          assignment => {
+            val duration = opts.options.valueOf(opts.resetByDurationOpt)
+            val durationParsed = Duration.parse(duration)
+            val instant = Instant.now().minus(durationParsed)
+            val timestamp = instant.toEpochMilli
+            val topicPartition = new TopicPartition(assignment.topic.get, assignment.partition.get)
+            val logTimestampOffset = getLogTimestampOffset(topicPartition, timestamp)
+            logTimestampOffset match {
+              case LogTimestampOffsetResult.LogTimestampOffset(offset) => assignment -> offset
+              case LogTimestampOffsetResult.Unknown => null
+              case LogTimestampOffsetResult.Ignore => null
+            }
+          }
+        ).toMap
+      } else if(opts.options.has(opts.resetFromFileOpt)){
+        val resetPlanPath = opts.options.valueOf(opts.resetFromFileOpt)
+        val resetPlanCsv = Utils.readFileAsString(resetPlanPath)
+        val resetPlan = parseResetPlan(resetPlanCsv)
+        assignmentsToReset.map(
+          assignment => {
+            val topicPartition = new TopicPartition(assignment.topic.get, assignment.partition.get)
+            if(resetPlan.keySet.contains(topicPartition)) assignment -> resetPlan(topicPartition)
+            else null
+          }
+        ).toMap
+      } else {
+        assignmentsToReset.map(
+          assignment => assignment -> assignment.offset.get
+        ).toMap
+      }
+    }
+
+    def resetAssignments(assignmentsToReset: Map[PartitionAssignmentState, Long]): Unit = {
+      val groupId = opts.options.valueOf(opts.groupOpt)
+      println(s"Resetting offsets from Consumer Group: $groupId")
+      assignmentsToReset.foreach {
+        case (assignment, offset) =>
+          val topicPartition = new TopicPartition(assignment.topic.get, assignment.partition.get)
+          consumer.assign(List(topicPartition).asJava)
+          consumer.seek(topicPartition, offset)
+          consumer.commitSync()
+          val newOffset = consumer.position(topicPartition)
```

Comment:
Done. I've change the `prepareAssignmentsToReset` return type to execute `commitSync(Map<TopicPartition, OffsetAndMetadata>)` directly.